packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,175 @@
+
+    ## User data fields.
+  - convert:
+      field: winlog.user_data.FileHashLength
+      type: long
+      ignore_missing: true
+      on_failure:
+        - remove:
+            field: winlog.user_data.FileHashLength
+            ignore_failure: true
+  - convert:
+      field: winlog.user_data.FilePathLength
+      type: long
+      ignore_missing: true
+      on_failure:
+        - remove:
+            field: winlog.user_data.FilePathLength
+            ignore_failure: true
+  - convert:
+      field: winlog.user_data.FqbnLength
+      type: long
+      ignore_missing: true
+      on_failure:
+        - remove:
+            field: winlog.user_data.FqbnLength
+            ignore_failure: true
+  - convert:
+      field: winlog.user_data.FullFilePathLength
+      type: long
+      ignore_missing: true
+      on_failure:
+        - remove:
+            field: winlog.user_data.FullFilePathLength
+            ignore_failure: true
+  - convert:
+      field: winlog.user_data.PolicyNameLength
+      type: long
+      ignore_missing: true
+      on_failure:
+        - remove:
+            field: winlog.user_data.PolicyNameLength
+            ignore_failure: true
+  - convert:
+      field: winlog.user_data.RuleNameLength
+      type: long
+      ignore_missing: true
+      on_failure:
+        - remove:
+            field: winlog.user_data.RuleNameLength
+            ignore_failure: true
+  - convert:
+      field: winlog.user_data.RuleSddlLength
+      type: long
+      ignore_missing: true
+      on_failure:
+        - remove:
+            field: winlog.user_data.RuleSddlLength
+            ignore_failure: true
+  - convert:
+      field: winlog.user_data.TargetProcessId
+      type: long
+      ignore_missing: true
+      on_failure:
+        - remove:
+            field: winlog.user_data.TargetProcessId
+            ignore_failure: true
+
+    ## Fully Qualified Binary Name (FQBN) / File Fields
+  - grok:
+      field: winlog.user_data.Fqbn
+      ignore_missing: true
+      patterns:
+      - "^CN=(?<file.x509.subject.common_name>.*),%{SPACE}O=(?<file.x509.subject.organization>.*),%{SPACE}L=(?<file.x509.subject.locality>.*),%{SPACE}S=(?<file.x509.subject.state_or_province>.*),%{SPACE}C=(?<file.x509.subject.country>[^\\\\]*)\\\\(?<file.pe.product>[^\\\\]*)\\\\(?<file.pe.original_file_name>[^\\\\]*)\\\\(?<file.pe.file_version>.*)$"
+      - "^CN=(?<file.x509.subject.common_name>.*),%{SPACE}O=(?<file.x509.subject.organization>.*),%{SPACE}L=(?<file.x509.subject.locality>.*),%{SPACE}C=(?<file.x509.subject.country>[^\\\\]*)\\\\(?<file.pe.product>[^\\\\]*)\\\\(?<file.pe.original_file_name>[^\\\\]*)\\\\(?<file.pe.file_version>.*)$"
+      - "^CN=(?<file.x509.subject.common_name>.*),%{SPACE}O=(?<file.x509.subject.organization>.*),%{SPACE}S=(?<file.x509.subject.state_or_province>.*),%{SPACE}C=(?<file.x509.subject.country>[^\\\\]*)\\\\(?<file.pe.product>[^\\\\]*)\\\\(?<file.pe.original_file_name>[^\\\\]*)\\\\(?<file.pe.file_version>.*)$"
```

```suggestion
      - '^CN=(?<file.x509.subject.common_name>.*),%{SPACE}O=(?<file.x509.subject.organization>.*),%{SPACE}L=(?<file.x509.subject.locality>.*),%{SPACE}S=(?<file.x509.subject.state_or_province>.*),%{SPACE}C=(?<file.x509.subject.country>[^\\]*)\\(?<file.pe.product>[^\\]*)\\(?<file.pe.original_file_name>[^\\]*)\\(?<file.pe.file_version>.*)$'
      - '^CN=(?<file.x509.subject.common_name>.*),%{SPACE}O=(?<file.x509.subject.organization>.*),%{SPACE}L=(?<file.x509.subject.locality>.*),%{SPACE}C=(?<file.x509.subject.country>[^\\]*)\\(?<file.pe.product>[^\\]*)\\(?<file.pe.original_file_name>[^\\]*)\\(?<file.pe.file_version>.*)$'
      - '^CN=(?<file.x509.subject.common_name>.*),%{SPACE}O=(?<file.x509.subject.organization>.*),%{SPACE}S=(?<file.x509.subject.state_or_province>.*),%{SPACE}C=(?<file.x509.subject.country>[^\\]*)\\(?<file.pe.product>[^\\]*)\\(?<file.pe.original_file_name>[^\\]*)\\(?<file.pe.file_version>.*)$'
```
(let's not proliferate `\`)

---

This makes sense. Curious to look back at the other integrations to see how we handled these.